modules/substrate/src/lib.rs

```diff
@@ -105,17 +107,32 @@ decl_storage! {
 		// Grandpa doesn't require there to always be a pending change. In fact, most of the time
 		// there will be no pending change available.
 		NextScheduledChange: map hasher(identity) BridgedBlockHash<T> => Option<ScheduledChange<BridgedBlockNumber<T>>>;
-		/// Whether or not the bridge has been initialized.
+		/// Optional pallet owner.
 		///
-		/// This is important to know to ensure that we don't try and initialize the bridge twice
-		/// and create an inconsistent genesis state.
-		IsInitialized: bool;
+		/// Pallet owner has a right to halt all pallet operations and then resume it. If it is
+		/// `None`, then there are no direct ways to halt/resume pallet operations, but other
+		/// runtime methods may still be used to do that (i.e. democracy::referendum to update halt
+		/// flag directly or call the `halt_operations`).
+		ModuleOwner get(fn module_owner): Option<T::AccountId>;
+		/// If true, all pallet transactions are failed immediately.
+		IsHalted get(fn is_halted) config(): bool;
 	}
 	add_extra_genesis {
+		config(owner): Option<T::AccountId>;
 		config(init_data): Option<InitializationData<BridgedHeader<T>>>;
 		build(|config| {
+			IsHalted::put(config.is_halted);
+
+			if let Some(ref owner) = config.owner {
+				<ModuleOwner<T>>::put(owner);
+			}
+
 			if let Some(init_data) = config.init_data.clone() {
 				initialize_bridge::<T>(init_data);
+			} else {
+				// Since the bridge hasn't been initialized we shouldn't allow anyone to perform
+				// transactions.
+				IsHalted::put(true);
```

Comment:
This means that we'll need to send two transactions in #483 to actually initialize + start pallet, right?

Reply:
(I mean - it is ok, just wanted to be sure that it is by design)

Reply:
IMHO this should be part of `init_data`, so that you can easily initialize & resume the bridge.

Reply:
> This means that we'll need to send two transactions in #483 to actually initialize + start pallet, right?

In its current form, yes that's correct